pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,408 @@
+	// part of Kubernetes's Container Spec.
+	STDIN     bool
+	STDINONCE bool
+	TTY       bool
+
+	// Linux contains configuration specific to Linux containers.
+	Linux *LinuxContainerConfig
+}
+
+// RawContainerStatus represents the status of a container.
+type RawContainerStatus struct {
+	// ID of the container.
+	ID ContainerID
+	// Name of the container.
+	Name string
+	// Status of the container.
+	State ContainerState
+	// Creation time of the container.
+	CreatedAt unversioned.Time
+	// Start time of the container.
+	StartedAt unversioned.Time
+	// Finish time of the container.
+	FinishedAt unversioned.Time
+	// Exit code of the container.
+	ExitCode int
+	// Reference to the image in use. For most runtimes, this should be an
+	// image ID.
+	ImageRef string
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map.
+	Annotations map[string]string
+	// A brief CamelCase string explains why container is in such a status.
+	Reason string
+}
+
+// LinuxContainerConfig contains platform-specific configuration for
+// Linux-based containers.
+type LinuxContainerConfig struct {
+	// Resources specification for the container.
+	Resources *LinuxContainerResources
+	// Capabilities to add or drop.
+	Capabilities *api.Capabilities
+	// SELinux is the SELinux context to be applied.
+	SELinux *api.SELinuxOptions
+	// TODO: Add support for seccomp.
+}
+
+// LinuxContainerResources specifies Linux specific configuration for
+// resources.
+// TODO: Consider using Resources from opencontainers/runtime-spec/specs-go
+// directly.
+type LinuxContainerResources struct {
+	// CPU CFS (Completely Fair Scheduler) period
+	CPUPeriod *int64
+	// CPU CFS (Completely Fair Scheduler) quota
+	CPUQuota *int64
+	// CPU shares (relative weight vs. other containers)
+	CPUShares *int64
+	// Memory limit in bytes
+	MemoryLimitInBytes *int64
+	// OOMScoreAdj adjusts the oom-killer score.
+	OOMScoreAdj *int64
+}
+
+// ContainerFilter is used to filter containers.
+type ContainerFilter struct {
+	// Name of the container.
+	Name *string
+	// ID of the container.
+	ID *RawContainerID
+	// State of the contianer.
+	State *ContainerState
+	// ID of the PodSandbox.
+	PodSandboxID *PodSandboxID
+	// LabelSelector to select matches.
+	// Only api.MatchLabels is supported for now and the requirements
+	// are ANDed. MatchExpressions is not supported yet.
+	LabelSelector unversioned.LabelSelector
+}
+
+type StreamOptions struct {
+	TTY          bool
+	InputStream  io.Reader
+	OutputStream io.Writer
+	ErrorStream  io.Writer
+}
+
+// KeyValue represents a key-value pair.
+type KeyValue struct {
+	Key   string
+	Value string
+}
+
+// ImageOperations offers basic image operations.
+type ImageOperations interface {
+	// List lists the existing images.
+	List() ([]Image, error)
+	// Pull pulls an image with authentication config.
+	Pull(image ImageSpec, auth AuthConfig) error
```

Comment:
should pull receive info such as connection information when actually pulling (e.g. cert, tls verification and such)